test/e2e/case12_ordering_test.go

```diff
@@ -0,0 +1,385 @@
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Warning",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"NonCompliant; there is violation")
+
+		By("Checking if dependency status is noncompliant")
+		Eventually(checkCompliance(case12DepName), defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+		By("Checking if policy status is pending")
+		Eventually(checkCompliance(case12PolicyName), defaultTimeoutSeconds, 1).Should(Equal("Pending"))
+	})
+	It("Should set to Compliant when dep status is resolved", func() {
+		By("Generating a non compliant event on the dep")
+		managedPlc := utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Warning",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"NonCompliant; there is violation")
+
+		By("Checking if dependency status is noncompliant")
+		Eventually(checkCompliance(case12DepName), defaultTimeoutSeconds, 1).Should(Equal("NonCompliant"))
+
+		By("Checking if policy status is pending")
+		Eventually(checkCompliance(case12PolicyName), defaultTimeoutSeconds, 1).Should(Equal("Pending"))
+
+		By("Generating an compliant event on the dependency")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12DepName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/namespace-foo-setup-configpolicy",
+			"Compliant; No violation detected")
+		By("Checking if dependency status is compliant")
+		Eventually(func() interface{} {
+			managedPlc = utils.GetWithTimeout(
+				clientManagedDynamic,
+				gvrPolicy,
+				case12DepName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+		By("Generating an compliant event on the policy")
+		managedPlc = utils.GetWithTimeout(
+			clientManagedDynamic,
+			gvrPolicy,
+			case12PolicyName,
+			clusterNamespace,
+			true,
+			defaultTimeoutSeconds)
+		Expect(managedPlc).NotTo(BeNil())
+		managedRecorder.Event(
+			managedPlc,
+			"Normal",
+			"policy: managed/case12-config-policy",
+			"Compliant; No violation detected")
+		By("Checking if policy status is compliant")
+		Eventually(func() interface{} {
+			managedPlc = utils.GetWithTimeout(
+				clientManagedDynamic,
+				gvrPolicy,
+				case12PolicyName,
+				clusterNamespace,
+				true,
+				defaultTimeoutSeconds)
+
+			return getCompliant(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+	})
+	It("Should remove template if dependency changes", func() {
+		By("Generating an compliant event on the dependency")
```

Comment:
Nit
```suggestion
		By("Generating a compliant event on the dependency")
```

there are a few spots like this